docs/api/js/utils.md

```diff
@@ -1,68 +1,563 @@
+### Gas
+
+#### `DEFAULT_GAS_PER_PUBDATA_LIMIT`
+
+- Use a large amount of gas for signing on layer 2.
+- The amount ensures any reasonable limit is accepted. 
 
-Utility method that returns the correctly formed `paymasterParams` object for the common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
+:::info
+- The operator is NOT required to use the actual value and can use any value up to that signed by the user.
+:::
 
 ```typescript
-export function getPaymasterParams(paymasterAddress: Address, paymasterInput: PaymasterInput): PaymasterParams
+export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
 ```
 
-The definition of the `PaymasterInput` can be found [here](./types.md).
+#### `REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT`
 
-## Useful gas constants
-Currently, there is no method to accurately estimate the required `gasPerPubdataLimit`. That's why for now, it is highly recommended to provide the `DEFAULT_GAS_PER_PUBDATA_LIMIT`. Users are not charged more by providing it.
-Later on it will be possible to query the current recommended limit.
+Any amount is possible for L1->L2 transactions because the cost per gas is adjusted.
 
-```typescript
-const GAS_PER_PUBDATA_BYTE = 17;
+```ts
+export const REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT = 800;
+```
 
-// The large L2 gas per pubdata to sign. This gas is enough to ensure that
-// any reasonable limit will be accepted. Note, that the operator is NOT required to
-// use the honest value of gas per pubdata and it can use any value up to the one signed by the user.
-// In the future releases, we will provide a way to estimate the current gasPerPubdata.
-export const DEFAULT_GAS_PER_PUBDATA_LIMIT = 50000;
+## Functions
+
+### `applyL1ToL2Alias`
+
+Converts the address that submitted a transaction to the inbox on L1 to the `msg.sender` viewed on L2.
+
+See also [`undol1tol2alias`](#undol1tol2alias).
+
+```ts
+export function applyL1ToL2Alias(address: string): string {
+    return ethers.utils.hexlify(ethers.BigNumber.from(address).add(L1_TO_L2_ALIAS_OFFSET));
+}
+```
+
+### `checkBaseCost`
+
+Checks the base cost is within prescribed limits.
+
+```ts
+export async function checkBaseCost(baseCost: ethers.BigNumber, value: ethers.BigNumberish | Promise<ethers.BigNumberish>) {
+    if (baseCost.gt(await value)) {
+        throw new Error(
+            `The base cost of performing the priority operation is higher than the provided value parameter ` +
+                `for the transaction: baseCost: ${baseCost}, provided value: ${value}`
+        );
+    }
+}
+```
+
+### `create2Address`
+
+Generates a future-proof contract address using salt plus bytecode which allows determination of an address before deployment. 
+
+:::warning
+- The zkSync Era implementation is slightly different from Ethereum.
+:::
+
+#### Inputs
+
+- `sender`: sender's address.
+- `bytecodeHash`: output from zkSolc.
+- `salt`: randomization element.
+- `input`: ABI encoded constructor arguments.
```

Comment:
please add this section for other functions too, or do not sure is anywhere so it would be unified

Reply:
OK, I have done this @AnastasiiaVashchuk  ... for any inputs I had a question, I have put ?? when I'm finished going through your comments, I'll ask you to check these again if I may? Thank you